### PR TITLE
Add doxygen documentation

### PR DESCRIPTION
**Pull request by aelog**
This PR adds doxygen documentation to all the main public classes and methods.

### CODE REVIEW

File: keychain.h [hunk 1]
```diff
@@ -45,23 +48,109 @@ class QKEYCHAIN_EXPORT Job : public QObject {
     QSettings* settings() const;
     void setSettings( QSettings* settings );
 
+    /**
+     * Call this method to start the job.
+     * Tipically you want to connect some slot to the finished() signal first.
```
Comment (frankosterfeld) [inline keychain.h]: Typically

File: keychain.h [hunk 2]
```diff
@@ -45,23 +48,109 @@ class QKEYCHAIN_EXPORT Job : public QObject {
...
+     * You can run the job either synchronously or asynchronously.
+     *
+     * In the first case you tipically use an inner event loop:
```
Comment (frankosterfeld) [inline keychain.h]: typically here, too

File: keychain.h [hunk 3]
```diff
@@ -45,23 +48,109 @@ class QKEYCHAIN_EXPORT Job : public QObject {
...
+     *
+     * \code
+     * SomeClass::startJob()
+     * {
+     *     QEventLoop eventLoop;
+     *     connect(job, &Job::finished, &eventLoop, &QEventLoop::quit);
+     *     job->start();
+     *     eventLoop.exec();
```
Comment (frankosterfeld) [inline keychain.h]: local event loops are evil, so I'd rather not suggest to use them. (as anything can happen before exec returns, and usually things one doesn't expect)

Comment (aelog) [inline keychain.h]: Fair enough (I mentioned them only because I saw they are used in `testclient.cpp` :p)

**Comment by frankosterfeld** [comment]
Thanks!

**Pull request merged**